drivers/gpio/gpio_stm32.c

```diff
@@ -369,6 +414,146 @@ static int gpio_stm32_read(struct device *dev, int access_op,
 	return 0;
 }
 
+static int gpio_stm32_port_get_raw(struct device *dev, u32_t *value)
+{
+	const struct gpio_stm32_config *cfg = dev->config->config_info;
+	GPIO_TypeDef *gpio = (GPIO_TypeDef *)cfg->base;
+
+	*value = LL_GPIO_ReadInputPort(gpio);
+
+	return 0;
+}
+
+static int gpio_stm32_port_set_masked_raw(struct device *dev, u32_t mask,
+					 u32_t value)
+{
+	const struct gpio_stm32_config *cfg = dev->config->config_info;
+	GPIO_TypeDef *gpio = (GPIO_TypeDef *)cfg->base;
+
+	LL_GPIO_WriteOutputPort(gpio, mask & value);
```

Comment:
This should be a read, modify, write operation.

Reply:
Right, thanks!